rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,446 @@
+- https://github.com/dhruvkb/monoverse/blob/main/ingestion_server/justfile
+- https://github.com/dhruvkb/monoverse/blob/main/load_testing/justfile
+
+**b:** see the following
+
+- https://github.com/dhruvkb/monoverse/blob/main/.gitignore
+- https://github.com/dhruvkb/monoverse/blob/main/api/.gitignore
+- https://github.com/dhruvkb/monoverse/blob/main/ingestion_server/.gitignore
+- https://github.com/dhruvkb/monoverse/blob/main/ingestion_server/test/.gitignore
+  (unchanged)
+- https://github.com/dhruvkb/monoverse/blob/main/load_testing/.gitignore
+  (unchanged)
+- https://github.com/dhruvkb/monoverse/blob/main/nginx/.gitignore (unchanged)
+
+#### Merging frontend
+
+Change workdir to WordPress/openverse-frontend repo.
+
+```bash
+$ mkdir frontend
+```
+
+Move everything into it except the following directories and files:
+
+| File / directory        | Reason                                 |
+| ----------------------- | -------------------------------------- |
+| .git/                   | For obvious reasons                    |
+| .github/                | Must be in the root                    |
+| .npmrc                  | For `pnpm` workspaces                  |
+| .pnpmfile.cjs           | For `pnpm` workspaces                  |
+| .editorconfig           | Useful across the monorepo             |
+| .eslintignore           | Read by ESLint running in pre-commit   |
+| .eslintrc.js            | Read by ESLint running in pre-commit   |
+| .prettierignore         | Read by Prettier running in pre-commit |
+| prettier.config.js      | Read by Prettier running in pre-commit |
+| .pre-commit-config.yaml | Read by pre-commit                     |
+| LICENSE                 | Root-level docs                        |
+| README.md               | Root-level docs                        |
+| CODE_OF_CONDUCT.md      | Root-level docs                        |
+| CONTRIBUTING.md         | Root-level docs                        |
+
+```bash
+$ git add . # Stage all renamed files
+$ git commit -m "Nest code under \`frontend/\`"
+```
+
+Switch to the monorepo.
+
+```bash
+$ git pull frontend main --allow-unrelated-histories
+```
+
+Since we fixed these files when merging the API, almost all these conflicts are
+redundant.
+
+| Expected conflicts      | Resolution                      |
+| ----------------------- | ------------------------------- |
+| .github/CODEOWNERS      | Use existing                    |
+| .prettierignore         | Use existing                    |
+| .pre-commit-config.yaml | Uncomment the commented regions |
+| .gitignore              | Use existing                    |
+| prettier.config.js      | Use existing                    |
+| README.md               | Use existing                    |
+| CONTRIBUTING.md         | Use existing                    |
+
+#### Housekeeping
+
+1. Create "stack: \*" labels to help with issue and PR management.
+   Spoiler/foreshadowing: these labels will be used for more things later.
```

Comment:
Perhaps this should happen _before_ the migration? If we created the labels in advance, we could then apply each label to _all issues_ in the respective repositories using the cli (I think this would be a very simple command to write), _before_ transferring. The labels would be preserved during the transfer.

Reply:
Yes, that makes more sense than transferring the issues, keeping a list of them and then applying labels.